pkg/agent/openflow/client.go

```diff
@@ -266,7 +273,20 @@ func (c *client) Initialize(roundNum uint64) (<-chan struct{}, error) {
 	// Ignore first notification, it is not a "reconnection".
 	<-connCh
 
-	c.cookieAllocator = cookie.NewAllocator(roundNum)
+	c.roundInfo = roundInfo
+	c.cookieAllocator = cookie.NewAllocator(roundInfo.RoundNum)
+
+	deleteExistingFlowsForRound := func() error {
+		cookieID, cookieMask := cookie.CookieMaskForRound(roundInfo.RoundNum)
+		return c.bridge.DeleteFlowsByCookie(cookieID, cookieMask)
+	}
+	// In the normal case, there should be no existing flows with the current round number. This
+	// is needed in case the agent was restarted before we had a chance to increment the round
+	// number (incrementing the round number happens once we are satisfied that stale flows from
+	// the previous round have been deleted).
+	if err := deleteExistingFlowsForRound(); err != nil {
```

Comment:
Perhaps `DeleteStaleFlows` can take roundNum as an argument and be named `DeleteFlowsByRoundNum` and be reused here.

Reply:
I didn't change the `DeleteStaleFlows` API as I want to keep the agent logic simpler (checking `prevRoundNum` is done in `DeleteStaleFlows`). However, I did introduce a new private method (`deleteFlowsByRoundNum`) to unify the deletion logic.